symfony/src/Controller/CommunicationController.php

```diff
@@ -382,6 +384,20 @@ public function rename(Request $request, Campaign $campaign, Communication $comm
         ]));
     }
 
+    /**
+     * @Route("campaign/{campaign}/communication/{communication}/relaunch", name="relaunch")
+     * @Security("is_granted('ROLE_ADMIN')")
+     * @param Communication $communication
+     * @param ProcessorInterface $processor
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function relaunchCommunication(Campaign $campaign, Communication $communication, ProcessorInterface $processor)
```

Comment:
you don't need all the parameters, you may only keep the communication in the path